terra/executor/resources.py

```diff
@@ -94,6 +97,19 @@ class Resource:
   file locking, it is easy to use and makes more sense to use lock files than
   other forms of IPC to track resources and handle seg faults. Soft file
   locking is also supported, but less preferred.
+
+  Note
+  ----
+  This class supports :ref:`executor`-s that use threading or multiprocessing
+  for spawning workers. There is no additional effort needed to maintain this
+  worker isolation. However, if a single non-multiprocess executor (e.g.
+  ``ThreadPoolExecutor``) worker were to spawn additional threads or processes
+  during a task's execution, then :class:`Resource` would have no way of
+  knowing that the new threads/processes were part of the same worker. The same
+  is true if a single multiprocess executor worker were to spawn additional
+  processes (additional threads would not be an issue). If this were to ever
```

Comment:
So a single non-multiprocess threadpoolexecutor worker spawning additional threads is **not** ok. but a single multiprocess threadpoolexecutor worker spawning additional threads **is** ok?